bee-common/src/logger/config.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright 2020 IOTA Stiftung
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and limitations under the License.
+
+use crate::logger::LOGGER_STDOUT_NAME;
+
+use log::LevelFilter;
+use serde::Deserialize;
+
+/// Default value for the color flag.
+const DEFAULT_COLOR: bool = true;
+/// Default name for an output.
+const DEFAULT_NAME: &str = LOGGER_STDOUT_NAME;
+/// Default log level for an output.
+const DEFAULT_LEVEL: &str = "info";
+
+/// Builder for a logger output configuration.
+#[derive(Default, Deserialize)]
+pub struct LoggerOutputConfigBuilder {
+    // Name of an output.
+    name: Option<String>,
+    // Log level of an output.
+    level: Option<String>,
+}
+
+impl LoggerOutputConfigBuilder {
+    /// Creates a new builder for a logger output configuration.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Sets the name of a logger output.
+    pub fn name(mut self, name: &str) -> Self {
+        self.name.replace(name.to_string());
+        self
+    }
```

Comment:
just curious: what does specifying the `name` allows us to do? Currently we just log to `stdout`, right? What if I'ld like to directly log to a file or some other stream?

edit: aaaah, this is actually a **file**name. That wasn't clear to me from the documentation. I only understood this by reading the code in `mod.rs`.

Reply:
It's an output name yes. Not always a **file**name.